### ISSUE TITLE
PostgreSQL constraint whitespace false positive

### ISSUE THREAD

**Issue by lhamt** [sqlfluff/sqlfluff#5211]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

I ran `sqlfluff --lint ./test.sql`

On this file (./test.sql):
```
DROP FUNCTION IF EXISTS F_NAME CASCADE;
```
And it reported the whitespace between the function name `F_NAME` and the keyword `CASCADE` as erroneous. The same happens if I use the constraint `RESTRICT` instead of `CASCADE`.

This is the first time I've used sqlfluff so I may be mistaken. Please correct me if I'm doing something incorrectly.

### Expected Behaviour

I expected it not to raise this as an error. I also would expect `sqlfluff fix` not to join the CASCADE key word to become part of the function name.
The PostgreSQL documentation for this query can be found [here](https://www.postgresql.org/docs/current/sql-dropfunction.html).

### Observed Behaviour

The command output after linting is as follows:
```
== [./test.sql] FAIL                                                                                                                                                                                                                                                     
L:   1 | P:  31 | LT01 | Unexpected whitespace before 'CASCADE' keyword.
                       | [layout.spacing]
All Finished 📜 🎉!
root@docker-desktop:/works
```
The content of the file after running `sqlfluff fix ./test.sql` is:
```
DROP FUNCTION IF EXISTS F_NAMECASCADE;
```
With `F_NAME` and `CASCADE` becoming one word `F_NAMECASCADE`.

### How to reproduce

1. Set config to what I have below (dialect = postgres and nothing else).
2. `echo "DROP FUNCTION IF EXISTS F_NAME CASCADE;" > test.sql`
3. sqlfluff lint ./test.sql

### Dialect

postgres

### Version

sqlfluff, version 2.3.2

### Configuration

[sqlfluff]
dialect = postgres

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by joaostorrer** [comment]
This is happening because the default value of `spacing_after` for `function_name` is `touch:inline`, so it doesn't allow spaces after the function name.
I'm willing to work on this, but I need some help to understand the best way to fix it.
@alanmcruickshank @greg-finley can you help ?

**Comment by alanmcruickshank** [comment]
It might be aityle but if a cheat, but I think the most pragmatic solution here would be to parse it as an identifier and not a function name. Then we won't get spacing issues.

Is it allowed to quote the function name too? If so then `ObjectReferenceSegment` is probably best.
